test/WebJobs.Script.Tests/FunctionMetadataManagerTests.cs

```diff
@@ -12,9 +12,12 @@
 using Microsoft.Azure.WebJobs.Script.Workers.Http;
 using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
+using Microsoft.WebJobs.Script.Tests;
 using Moq;
 using Xunit;
+using static Microsoft.Azure.AppService.Proxy.Common.Constants.WellKnownHttpHeaders;
```

Comment:
I see there's also a FunctionMetadataProviderTests test suite that exercises WorkerIndexing. I think you should add/update some tests there as well so we have test coverage for all 3 provider flows you're touching here.

Reply:
Changed!